### DISCUSSION TITLE
Accessing an object

### DISCUSSION THREAD

**Discussion by Mikhail666999** [gilzoide/godot-lua-pluginscript#29] [Q&A] [answered]
Hello, how to access the object?
in gdscript:
`onready var Lof = $lof`

and in lua how?

**Comment by gilzoide** [comment, marked as answer]
Hi @Mikhail666999 !
When you use the `onready` keyword for setting a script variable, it's the same as setting it inside the script's `_ready` method (docs [here](https://docs.godotengine.org/en/3.3/getting_started/scripting/gdscript/gdscript_basics.html#literals)).
Also, the literal `$lof` is the same as calling `get_node("lof")` (docs [here](https://docs.godotengine.org/en/3.3/getting_started/scripting/gdscript/gdscript_basics.html#literals)).

So in Lua all you need to do is set the variable inside the `_ready` method using `get_node`, which is exactly what GDScript does in your snippet:
```lua
local MyClass = {
  extends = 'Node',
}

function MyClass:_ready()
  -- `get_node` must be called from `self`, as it is a method from the Node class
  self.Lof = self:get_node("lof")
end

-- ...

return MyClass
```

**Reply by TrainzMarcel** [reply]
please please add this to the script example, i had to look all over the place and still couldnt find out how to access other nodes. the documentation doesnt make this very clear at all 🥺

**Reply by gilzoide** [reply]
Hey @TrainzMarcel, thanks for the feedback.
Yeah, there needs to be a "From GDScript to Lua" page in the documentation with these things, I'm really sorry about not having it already. To be honest, I can't remember why there isn't one in the first place 🤔
I'll try to work on one this weekend, thanks again!

**Comment by Mikhail666999** [comment]
understood thanks)

**Comment by gilzoide** [comment]
Hey folks, so I've added a [From GDScript to Lua](https://github.com/gilzoide/godot-lua-pluginscript/blob/main/extras/docs/from-gdscript-to-lua.md) document to the repo!
It's a work in progress for better documentation on how to use Lua PluginScript.

It's far from a comprehensive list, but it's a start. Throughout the following weeks (most likely weekends, most likely 2023 only) I want to improve this document and create any others that might make sense =]

Thanks again for your feedbacks, feel free to keep in touch!
